### PR TITLE
Improve the onboarding access checklist for contributors, maintainers and administrators

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fixes #

**Preview**: https://pygmt-dev--2656.org.readthedocs.build/en/2656/maintenance.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: doc/maintenance.md [hunk 1]
```diff
@@ -9,12 +9,23 @@ instead.
 
 ## Onboarding Access Checklist
 
-- Added to [python-maintainers](https://github.com/orgs/GenericMappingTools/teams/python-maintainers) team in the [GenericMappingTools](https://github.com/orgs/GenericMappingTools/teams/) organization on GitHub (gives 'maintain' permissions)
+### As a Contributor
+
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:13]: I would move the "pygmt-contribuors team" point to the top:
```suggestion

- Added to the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)
```

File: doc/maintenance.md [hunk 2]
```diff
@@ -9,12 +9,23 @@ instead.
...
 - Added as collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)
-- Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
+- Added to [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:15]:
```suggestion

```

File: doc/maintenance.md [hunk 3]
```diff
@@ -9,12 +9,23 @@ instead.
...
 - Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
+- Added as member on [HackMD](https://hackmd.io/@pygmt) (for draft annoucements) [optional]
+
+### As a Maintainer
+
+- Added to [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:21]:
```suggestion
- Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
```

File: doc/maintenance.md [hunk 4]
```diff
@@ -9,12 +9,23 @@ instead.
...
+- Update the roles in the {doc}`Team Gallery <team>` page
+- Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
+
+### As an Administrator
+
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:26]: I would the "pygmt-admin team" point to the top:
```suggestion

- Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)
```

File: doc/maintenance.md [hunk 5]
```diff
@@ -9,12 +9,23 @@ instead.
...
 - Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]
-- Added as member on [HackMD](https://hackmd.io/@pygmt) [optional]
+- Added to [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) (gives 'admin' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:28]:
```suggestion

```

File: doc/maintenance.md [hunk 6]
```diff
@@ -9,12 +9,23 @@ instead.
...
+### As a Maintainer
+
+- Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
+- Update the roles in the {doc}`Team Gallery <team>` page
+- Added as moderator on the [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions) [optional]
+
+### As an Administrator
+
+- Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)
 - Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]
```
Comment (weiji14) [inline doc/maintenance.md:28]: Include admin permissons on DAGsHub too? Mark as optional or not?

```suggestion
- Added as admin on [DAGsHub](https://www.dagshub.com/GenericMappingTools/pygmt/settings/collaboration)
- Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]
```

**Review by michaelgrund — APPROVED** [review]

File: doc/maintenance.md [hunk 7]
```diff
@@ -9,12 +9,30 @@ instead.
...
 
-- Added to [python-maintainers](https://github.com/orgs/GenericMappingTools/teams/python-maintainers) team in the [GenericMappingTools](https://github.com/orgs/GenericMappingTools/teams/) organization on GitHub (gives 'maintain' permissions)
+Note that anyone can contribute to PyGMT, even without being added to the
+[GenericMappingTools team](https://github.com/orgs/GenericMappingTools/teams).
+The onboarding items below are for people who would like to make regular
+contributions, and could benefit from extra permissions to the developer and
+communication tools we use.
+### As a Contributor
+
+- Added to the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:19]: For consistency with line 13, I would include "team" in the link.
```suggestion
- Added to the [pygmt-contributors team](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) (gives 'write' permission to the repository)
```

File: doc/maintenance.md [hunk 8]
```diff
@@ -9,12 +9,30 @@ instead.
...
 - Added as collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)
-- Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
 - Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
+- Added as member on [HackMD](https://hackmd.io/@pygmt) (for draft announcements) [optional]
+
+### As a Maintainer
+
+- Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:26]:
```suggestion
- Added to the [pygmt-maintainers team](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) (gives 'maintain' permission to the repository)
```

File: doc/maintenance.md [hunk 9]
```diff
@@ -9,12 +9,30 @@ instead.
...
+- Update the roles in the {doc}`Team Gallery <team>` page
+- Added as moderator on the [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions) [optional]
+- Added as a maintainer on [Readthedocs](https://readthedocs.org/projects/pygmt-dev) [optional]
+
+### As an Administrator
+
+- Added to the [pygmt-admin](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) team (gives 'admin' permission to the repository)
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:33]:
```suggestion
- Added to the [pygmt-admin team](https://github.com/orgs/GenericMappingTools/teams/pygmt-admin) (gives 'admin' permission to the repository)
```

File: doc/maintenance.md [hunk 10]
```diff
@@ -9,12 +9,30 @@ instead.
...
+- Added to the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team (gives 'write' permission to the repository)
 - Added as collaborator on [DAGsHub](https://dagshub.com/GenericMappingTools/pygmt/settings/collaboration) (gives 'write' permission to dvc remote storage)
-- Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
 - Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
+- Added as member on [HackMD](https://hackmd.io/@pygmt) (for draft announcements) [optional]
+
+### As a Maintainer
+
+- Added to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team (gives 'maintain' permission to the repository)
+- Update the roles in the {doc}`Team Gallery <team>` page
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:27]: Not completly sure here, but maybe singular makes for sense here?

```suggestion
- Update the role on the {doc}`Team Gallery page <team>`
```

**Review by yvonnefroehlich — APPROVED** [review]

File: doc/maintenance.md [hunk 11]
```diff
@@ -9,12 +9,30 @@ instead.
 
 ## Onboarding Access Checklist
 
-- Added to [python-maintainers](https://github.com/orgs/GenericMappingTools/teams/python-maintainers) team in the [GenericMappingTools](https://github.com/orgs/GenericMappingTools/teams/) organization on GitHub (gives 'maintain' permissions)
+Note that anyone can contribute to PyGMT, even without being added to the
+[GenericMappingTools team](https://github.com/orgs/GenericMappingTools/teams).
+The onboarding items below are for people who would like to make regular
+contributions, and could benefit from extra permissions to the developer and
+communication tools we use.
+### As a Contributor
```
Comment (weiji14) [inline doc/maintenance.md:17]:
```suggestion
communication tools we use.

### As a Contributor
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**